app/src/main/java/ch/epfl/sdp/drone3d/ui/mission/ItineraryShowActivity.kt

```diff
@@ -85,55 +85,58 @@ class ItineraryShowActivity : BaseMapActivity() {
                     missionDrawer = MapboxMissionDrawer(mapView, mapboxMap, mapboxMap.style!!)
                 }
 
-                if (currentMissionPath != null) {
-                    missionDrawer.showMission(currentMissionPath!!, false)
-                    MapboxUtility.zoomOnMission(currentMissionPath!!, mapboxMap)
+                val missionBuilder = ParallelogramMappingMissionService(droneService)
+                flightPath = when (strategy) {
+                    MappingMissionService.Strategy.SINGLE_PASS -> missionBuilder.buildSinglePassMappingMission(
+                        area,
+                        flightHeight
+                    )
+                    MappingMissionService.Strategy.DOUBLE_PASS -> missionBuilder.buildDoublePassMappingMission(
+                        area,
+                        flightHeight
+                    )
                 }
+                missionDrawer.showMission(flightPath, false)
+                MapboxUtility.zoomOnMission(flightPath, mapboxMap)
             }
         }
 
         findViewById<View>(R.id.mission_delete).visibility =
             if (authService.getCurrentSession()?.user?.uid == ownerUid) View.VISIBLE else View.GONE
 
-        if (currentMissionPath != null && currentMissionPath!!.isNotEmpty()) {
-            weatherReport = weatherService.getWeatherReport(LatLng(currentMissionPath!![0]))
-            weatherReport.observe(this) {
-                isWeatherGoodEnough = WeatherUtils.isWeatherGoodEnough(it)
-            }
+        weatherReport = weatherService.getWeatherReport(area[0])
+        weatherReport.observe(this) {
+            isWeatherGoodEnough = WeatherUtils.isWeatherGoodEnough(it)
         }
     }
 
-    override fun onResume() {
-        super.onResume()
-        droneService.getData().isConnected().observe(this) {
-            findViewById<View>(R.id.buttonToMissionInProgressActivity).isEnabled = canMissionBeLaunched()
-        }
-    }
-
-    override fun onPause() {
-        super.onPause()
-
-        droneService.getData().isConnected().removeObservers(this)
-    }
-
-    /**
-     * Check if there is a connected drone, and if the user or the simulation is close enough to launch a mission
-     */
-    private fun canMissionBeLaunched(): Boolean {
-        val dronePos = droneService.getData().getPosition().value
-        return if (currentMissionPath?.isEmpty() == true || !droneService.isConnected() || dronePos == null)
-            false
-        else {
-            val beginningPoint = currentMissionPath!![0]
-            dronePos.distanceTo(beginningPoint) < MAX_BEGINNING_DISTANCE
+    private fun readIntent() {
```

Comment:
I don't think this name is the best to describe this function. I'd suggest something along the line of extractExtras() or getExtras()